xla/backends/gpu/codegen/custom.cc

```diff
@@ -547,44 +540,50 @@ absl::StatusOr<FusionEmissionResult> EmitGemm(
   std::optional<HloInstruction*> while_op =
       GetParentWhileOp(fusion, call_graph);
   std::unique_ptr<HloModule> init_module, update_module;
+  std::optional<int64_t> indvar_idx = std::nullopt;
   if (while_op != std::nullopt) {
     CHECK(while_op.value() != nullptr)
         << "GetWhileOp is not expected to return nullptr.";
-    init_module = ExtractWhileInitModule(*while_op);
-    update_module = ExtractWhileUpdateModule(*while_op);
+    indvar_idx = GetLoopInductionVarTupleIdx(*while_op);
+    if (indvar_idx != std::nullopt) {
+      init_module = ExtractWhileInitModule(*while_op, indvar_idx.value());
+      update_module = ExtractWhileUpdateModule(*while_op, indvar_idx.value());
+    }
   }
   bool can_compute_indvar_on_host =
       (init_module != nullptr && update_module != nullptr);
 
   unsigned arg_idx = 0;
-  TF_ASSIGN_OR_RETURN(BufferAllocation::Slice lhs_slice,
-                      GetOperandSlice(buffer_assignment, adaptor, fusion,
-                                      *custom_call.operand(arg_idx),
-                                      slice_instrs, /*shape_idx=*/{}, arg_idx));
+  TF_ASSIGN_OR_RETURN(
```

Comment:
Unless I am missing something, I think this didn't change (?). In the future, please don't change the formatting of lines you didn't touch. This makes it harder to understand what actually changed (and pollutes the history).